lib/src/components/NxDrawer/NxDrawer.scss

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+ @use '../../scss-shared/nx-container-helpers';
+ @use '../../scss-shared/nx-text-helpers';
+ 
+$-box-shadow-offset: 52px;
+
+.nx-drawer {
+  // Undo styles that chrome applies to <dialog> elements
+  border: initial;
+  height: initial;
+  left: initial;
+  margin: initial;
+  max-height: initial;
+  max-width: initial;
+  padding: initial;
+  width: initial;
+
+  // Prevent brief flash of scrollbar as tooltips disappear
+  overflow: hidden;
+
+  &::backdrop {
+    // Chrome tries to add its own subtle mask here which doubles up with our own
+    display: none;
+    pointer-events: auto;
+  }
+
+  background: transparent;
+  box-sizing: border-box;
+  height: 100%;
+  padding-left: $-box-shadow-offset;
+  position: fixed;
+  right: 0;
+  top: 0;
+  transform: translateX(100%);
+  width: 520px + $-box-shadow-offset;
+  z-index: 100;
+
+  &--with-global-header {
+    $global-header-height: 43px;
+    height: calc(100% - #{$global-header-height});
+    top: $global-header-height;
+  }
+  
+  &--narrow {
+    width: 348px + $-box-shadow-offset;
```

Comment:
I take the adjustment here is needed so that the overflow doesn't cut off the shadow? That's another thing that could be simplified if the `<dialog>` was implemented to be a full-screen mask as I described in another comment.

Reply:
> I take the adjustment here is needed so that the overflow doesn't cut off the shadow? That's another thing that could be simplified if the `<dialog>` was implemented to be a full-screen mask as I described in another comment.

We tried implementing dialog to be a full screen mask and it worked great but after switching to the nx-page layout and making the area outside of the drawer interactive we went back to the prev impl.

Reply:
> We tried implementing dialog to be a full screen mask and it worked great but after switching to the nx-page layout and making the area outside of the drawer interactive we went back to the prev impl.

AFAICT this extra space in the `.nx-drawer` and the margin on the `.nx-drawer__panel` are present to make the box-shadow visible.  The reason the box-shadow isn't visible without that is because the `.nx-drawer` has `overflow: hidden`.  As far as I can tell, the only reason that is has `overflow: hidden` is to keep its height constrained when its contents are long - otherwise it expands the height of the entire `content` grid area.  There is another better way to fix that however: by setting `min-height: 0` on the `.nx-drawer`.  Once you do that, unless I missed another reason for it, you can remove the `overflow: hidden` and then remove the gutter space for the shadow.

Reply:
> > We tried implementing dialog to be a full screen mask and it worked great but after switching to the nx-page layout and making the area outside of the drawer interactive we went back to the prev impl.
> 
> AFAICT this extra space in the `.nx-drawer` and the margin on the `.nx-drawer__panel` are present to make the box-shadow visible. The reason the box-shadow isn't visible without that is because the `.nx-drawer` has `overflow: hidden`. As far as I can tell, the only reason that is has `overflow: hidden` is to keep its height constrained when its contents are long - otherwise it expands the height of the entire `content` grid area. There is another better way to fix that however: by setting `min-height: 0` on the `.nx-drawer`. Once you do that, unless I missed another reason for it, you can remove the `overflow: hidden` and then remove the gutter space for the shadow.

Ah I think the other reason for the `overflow-hidden` is so that when the drawer is being translated (as it animates) it doesn't create a horizontal scroll bar at the bottom.

Reply:
> Ah I think the other reason for the overflow-hidden is so that when the drawer is being translated (as it animates) it doesn't create a horizontal scroll bar at the bottom.

I'd say the right fix for that particular problem would be `overflow: hidden` on `.nx-page`.  Unless that turns out to cause other problems.

Reply:
> > Ah I think the other reason for the overflow-hidden is so that when the drawer is being translated (as it animates) it doesn't create a horizontal scroll bar at the bottom.
> 
> I'd say the right fix for that particular problem would be `overflow: hidden` on `.nx-page`. Unless that turns out to cause other problems.

Updated here: ecc15ed61c3bc863a7f20fa4b5b100a6472570cd

Reply:
> Updated here: https://github.com/sonatype/sonatype-react-shared-components/commit/ecc15ed61c3bc863a7f20fa4b5b100a6472570cd

That partially does what I was suggesting, but not all of it and also adds other complications that I don't think are necessary.  Specifically, `overflow: hidden` has been added to `.nx-page` but it has not been removed from `.nx-drawer`.  As a result, `.nx-drawer` is still having to be wider than the visible drawer itself - in fact it seems that it now spans the entire viewport.  And then to allow clicks to pass through it, we're having to mess with `pointer-events`.  I don't think all of that is necessary - if `overflow: hidden` is removed from `.nx-drawer` I think its size can be constrained to its visible area.  Then once that's done, we can move on to the question of whether we really need all three of `.nx-drawer`, `.nx-drawer__animation-wrapper`, and `.nx-drawer__panel`.  I suspect one or possibly two of those can be eliminated.

Reply:
> > Updated here: [ecc15ed](https://github.com/sonatype/sonatype-react-shared-components/commit/ecc15ed61c3bc863a7f20fa4b5b100a6472570cd)
> 
> That partially does what I was suggesting, but not all of it and also adds other complications that I don't think are necessary. Specifically, `overflow: hidden` has been added to `.nx-page` but it has not been removed from `.nx-drawer`. As a result, `.nx-drawer` is still having to be wider than the visible drawer itself - in fact it seems that it now spans the entire viewport. And then to allow clicks to pass through it, we're having to mess with `pointer-events`. I don't think all of that is necessary - if `overflow: hidden` is removed from `.nx-drawer` I think its size can be constrained to its visible area. Then once that's done, we can move on to the question of whether we really need all three of `.nx-drawer`, `.nx-drawer__animation-wrapper`, and `.nx-drawer__panel`. I suspect one or possibly two of those can be eliminated.

We cleaned it up here: c784fa4f0e1221ae3e12ab5ccfe45d17298289ed

Reply:
Oh we couldn't get rid of panel because of issues with the z-index stacking order (the shadows would pop up when it is being animated because of the transform)

Reply:
> > Updated here: [ecc15ed](https://github.com/sonatype/sonatype-react-shared-components/commit/ecc15ed61c3bc863a7f20fa4b5b100a6472570cd)
> 
> That partially does what I was suggesting, but not all of it and also adds other complications that I don't think are necessary. Specifically, `overflow: hidden` has been added to `.nx-page` but it has not been removed from `.nx-drawer`. As a result, `.nx-drawer` is still having to be wider than the visible drawer itself - in fact it seems that it now spans the entire viewport. And then to allow clicks to pass through it, we're having to mess with `pointer-events`. I don't think all of that is necessary - if `overflow: hidden` is removed from `.nx-drawer` I think its size can be constrained to its visible area. Then once that's done, we can move on to the question of whether we really need all three of `.nx-drawer`, `.nx-drawer__animation-wrapper`, and `.nx-drawer__panel`. I suspect one or possibly two of those can be eliminated.

We found that by adding overflow: hidden to NxPage the gallery header no longer sticks to the top, breaking our visual tests. Should we find a way to override the implementation of the gallery header so that it works as before or do you foresee other potential side effects?

Reply:
> We found that by adding overflow: hidden to NxPage the gallery header no longer sticks to the top, breaking our visual tests. Should we find a way to override the implementation of the gallery header so that it works as before or do you foresee other potential side effects?

This only matters for page-scrolling legacy page layouts correct? Those aren't compatible with `NxDrawer` anyway, so you can adjust the styling for that case independently of the styling we add to facilitate drawers.  That is, you can add `overflow: hidden` to `.nx-page` and then add `overflow: initial;` to `.nx-html--page-scrolling .nx-page`.

Reply:
> Oh we couldn't get rid of panel because of issues with the z-index stacking order (the shadows would pop up when it is being animated because of the transform)

I see, maybe.  I'm seeing an issue with the shadow but it doesn't have to do with the animation, it's because the `::after` and the drawer contents are all siblings, and the drawer contents have no background of their own so you can see the `::after` behind them.  One way to fix that is to also use `::before`, to create just a white rectangle in front of the shadow but behind the other contents.

Reply:
> I see, maybe. I'm seeing an issue with the shadow but it doesn't have to do with the animation, it's because the ::after and the drawer contents are all siblings, and the drawer contents have no background of their own so you can see the ::after behind them. One way to fix that is to also use ::before, to create just a white rectangle in front of the shadow but behind the other contents.

But actually, I'm still skeptical that this shadow needs to be implemented as a separate element at all.  It should be possible to manipulate the spread radius and x-offset of the shadow to achieve exactly the desired effect using a `box-shadow` directly on the `nx-drawer`.

Reply:
> But actually, I'm still skeptical that this shadow needs to be implemented as a separate element at all. It should be possible to manipulate the spread radius and x-offset of the shadow to achieve exactly the desired effect using a box-shadow directly on the nx-drawer.

Specifically:

The shadow prescribed by Design has a x offset of -2, a y offset of 0, a blur of 50 and a spread of zero.  But it's applied to an element that is actually 64px shorter than the element that the shadow is perceived to be coming from.  So what I believe we can do is use a spread radius of -32px (64 / 2) to decrease the height of the shadow box to the correct size.  This will also decrease its width, so we have to cancel that out by moving the shadow box farther to the left: 32px farther than what design specified, for a total x-offset of -34px.  The right end of the shadow will be shorter, but that doesn't matter because we can't see it.  So overall, I believe a `box-shadow: -34px 0 50px -32px hsla(0, 0%, 50%, 0.5)` directly on the `.nx-drawer` should allow for the removal of the `::after`.

Side note: why is your current `box-shadow` color written in `rgba` syntax?

Reply:
> > We found that by adding overflow: hidden to NxPage the gallery header no longer sticks to the top, breaking our visual tests. Should we find a way to override the implementation of the gallery header so that it works as before or do you foresee other potential side effects?
> 
> This only matters for page-scrolling legacy page layouts correct? Those aren't compatible with `NxDrawer` anyway, so you can adjust the styling for that case independently of the styling we add to facilitate drawers. That is, you can add `overflow: hidden` to `.nx-page` and then add `overflow: initial;` to `.nx-html--page-scrolling .nx-page`.

Fixed here: 373300714b5048f0f2cd336d7258d880128916dc

Reply:
> > But actually, I'm still skeptical that this shadow needs to be implemented as a separate element at all. It should be possible to manipulate the spread radius and x-offset of the shadow to achieve exactly the desired effect using a box-shadow directly on the nx-drawer.
> 
> Specifically:
> 
> The shadow prescribed by Design has a x offset of -2, a y offset of 0, a blur of 50 and a spread of zero. But it's applied to an element that is actually 64px shorter than the element that the shadow is perceived to be coming from. So what I believe we can do is use a spread radius of -32px (64 / 2) to decrease the height of the shadow box to the correct size. This will also decrease its width, so we have to cancel that out by moving the shadow box farther to the left: 32px farther than what design specified, for a total x-offset of -34px. The right end of the shadow will be shorter, but that doesn't matter because we can't see it. So overall, I believe a `box-shadow: -34px 0 50px -32px hsla(0, 0%, 50%, 0.5)` directly on the `.nx-drawer` should allow for the removal of the `::after`.
> 
> Side note: why is your current `box-shadow` color written in `rgba` syntax?

Ooo that worked Thanks! We got rid of panel as per your suggestion:
7e383e3290f9e6615b04f1df8805eda0730c9aac

(We were using rgba because it was copy pasted from Sketch but we forgot to change it to hsla).